.github/workflows/docker.yml

```diff
@@ -0,0 +1,81 @@
+# The present workflow was made based on the following references:
+# - https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching
+# - https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+---
+name: Docker
+
+on: push
+
+concurrency:
+  group: docker-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx-
+          path: /tmp/.buildx-cache
+      - uses: docker/build-push-action@v2
+        with:
+          context: .
+          tags: ${{ env.IMAGE_NAME }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+          load: true
+      - run: docker run ${{ env.IMAGE_NAME }} --version
+      - run: |
+          # this is a workaround, check the following issues:
+          # - https://github.com/docker/build-push-action/issues/252
+          # - https://github.com/moby/buildkit/issues/1896
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+  publish:
+    if: contains(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    needs: [build]
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx-
+          path: /tmp/.buildx-cache
+      - uses: docker/login-action@v1 
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ${{ env.REGISTRY }}
+      - uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        id: meta
+      - uses: docker/build-push-action@v2
+        with:
+          context: .
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+          push: true
+      - run: |
+          # this is a workaround, check the following issues:
+          # - https://github.com/docker/build-push-action/issues/252
+          # - https://github.com/moby/buildkit/issues/1896
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
```

Comment:
@sestrella I see some duplicated steps between `build` and `publish`. Is it possible to use [node anchors](https://en.wikipedia.org/wiki/YAML#Advanced_components) and references (or [merge](https://yaml.org/type/merge.html)) to make it DRY?

Reply:
@CristhianMotoche unfortunately, YAML anchors are [not supported](https://github.community/t/support-for-yaml-anchors/16128) yet, on the other hand, there is something called [composite run steps](https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/) but it involves setting up a new repo which adds a level of indirection, so I think some code duplication in the context of this workflow is OK. What do you think?

Reply:
Too bad anchors are not supported yet. I don't think we should complicate things with the _composite run steps_. Let's keep it simple.